File: frontend/src/services/claimService.js
import api from './api'

export const createClaim = async (claimData) => {
  const response = await api.post('/claim/create', claimData)
  return response.data
}

export const getClaimById = async (id) => {
  const response = await api.get(`/claim/${id}`)
  return response.data
}

export const getMyClaims = async () => {
  const response = await api.get('/claim/my')
  return response.data
}

export const getReceivedClaims = async () => {
  const response = await api.get('/claim/received')
  return response.data
}

export const approveClaim = async (id) => {
  const response = await api.put(`/claim/${id}/approve`)
  return response.data
}

export const rejectClaim = async (id) => {
  const response = await api.put(`/claim/${id}/reject`)
  return response.data
}

File: LOST-FOUND/claim_service/src/main/java/com/lostfound/claim_service/service/ClaimService.java
package com.lostfound.claim_service.service;

import com.lostfound.claim_service.dto.ClaimRequest;
import com.lostfound.claim_service.dto.Response;

public interface ClaimService {

    Response createClaim(ClaimRequest request, String token);

    Response getClaimById(Long id);

    Response getMyClaims(Long claimedBy);

    Response getReceivedClaims(Long foundItemOwnerId);

    Response approveClaim(Long id, Long foundItemOwnerId);

    Response rejectClaim(Long id, Long foundItemOwnerId);

}
